### PR TITLE
Tock 2.0: touch api and example app for 2.0 API

### PR DESCRIPTION
**Pull request by alexandruradovici**
This pull request adds the touch driver API and example.

It should close #126

### CODE REVIEW

File: libtock/touch.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          free (multi_touch_buffer);
+          multi_touch_buffer = NULL;
+        }
+      }else {
+        err = TOCK_ENOMEM;
+      }
+    }else {
+      err = TOCK_EALREADY;
+    }
+  }else {
```
Comment (phil-levis) [inline libtock/touch.c:74]: Can you run `format_all.sh` in `examples/` to format the code?

File: libtock/touch.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#define GESTURE_SWIPE_RIGHT 4
+#define GESTURE_ZOOM_IN 5
+#define GESTURE_ZOOM_OUT 6
+
+typedef void (touch_callback)(int, int, int, void*);
+typedef void (gesture_callback)(int, void*);
+
+// buffer data format
+//  0         1           2                  4                  6           7             8         ...
+// +---------+-----------+------------------+------------------+-----------+---------------+--------- ...
```
Comment (phil-levis) [inline libtock/touch.h:26]: Can you define a structure with this format? Hardcoded offsets are buggy.

**Review by phil-levis — CHANGES_REQUESTED** [review]

File: libtock/touch.h (hunk 2 above)
Comment (alexandruradovici) [reply libtock/touch.h:26]: Your suggestion would be to use a packed structure instead of the buffer?

File: libtock/touch.c (hunk 1 above)
Comment (alexandruradovici) [reply libtock/touch.c:74]: I did, this is what it does. I actually ran `make format`. I fixed the other `}else  {` manually, but I missed this one :)

**Review by phil-levis — APPROVED** [review]

**Pull request merged**